src/VisualStudio/Core/Def/Implementation/ProjectSystem/MiscellaneousFilesWorkspace.cs

```diff
@@ -459,13 +382,66 @@ private void DetachFromDocument(uint docCookie, string moniker)
             }
         }
 
+        private void SubscribeToRunningDocTableEvents()
+        {
+            _runningDocumentTableEventSink.OnRenameDocument += HandleRenameDocumentEvent;
+            _runningDocumentTableEventSink.OnReloadDocumentData += HandleReloadDocumentEvent;
+            _runningDocumentTableEventSink.OnInitializedDocument += HandleInitializedDocumentEvent;
+            _runningDocumentTableEventSink.OnCloseDocument += HandleCloseDocumentEvent;
+        }
+
+        private void UnSubscribeFromRunningDocTableEvents()
+        {
+            _runningDocumentTableEventSink.OnRenameDocument -= HandleRenameDocumentEvent;
+            _runningDocumentTableEventSink.OnReloadDocumentData -= HandleReloadDocumentEvent;
+            _runningDocumentTableEventSink.OnInitializedDocument -= HandleInitializedDocumentEvent;
+            _runningDocumentTableEventSink.OnCloseDocument -= HandleCloseDocumentEvent;
+        }
+
+        private void HandleRenameDocumentEvent(object sender, RunningDocumentTableRenamedEventArgs args)
+        {
+            // We want to consider this file to be added in one of two situations:
+            //
+            // 1) the old file already was a misc file, at which point we might just be doing a rename from
+            //    one name to another with the same extension
+            // 2) the old file was a different extension that we weren't tracking, which may have now changed
+            if (TryUntrackClosingDocument(args.DocCookie, args.OldMoniker) || TryGetLanguageInformation(args.OldMoniker) == null)
+            {
+                if (_runningDocumentTableEventSink.TryGetBuffer(args.DocCookie, out var buffer))
```

Comment:
feels like " if (TryUntrackClosingDocument(args.DocCookie, args.OldMoniker) || TryGetLanguageInformation(args.OldMoniker) == null)" is not necessary?

could it be just

TryUntrackClosingDocument(args.DocCookie, args.OldMoniker);
 if (_runningDocumentTableEventSink.TryGetBuffer(args.DocCookie, out var buffer))
{
 TrackOpenedDocument(args.DocCookie, args.Moniker, buffer);
}

..

TrackOpenDocument seems already does this

var languageInformation = TryGetLanguageInformation(moniker); 
if (languageInformation == null) 
{ 
                // We can never put this document in a workspace, so just bail 
               return;
}

Reply:
Trackopendocument does have the check, but it's getting invoked on the new moniker whereas the check in the original if statement is getting invoked on the old moniker